library/HelloSign/TemplateSignatureRequest.php

```diff
@@ -113,15 +112,32 @@ public function setCC($role, $email)
     }
 
     /**
-     * Set the value to fill in for a custom field with the given field name
+     * Add value in the custom fields
      *
      * @param  string $field_name field name to be filled in
      * @param  string $value
+     * @param  string $editor     Signer
+     * @param  string $required   (true to allow Editable Merge, only supported for single signer requests)
      * @return TemplateSignatureRequest
      */
-    public function setCustomFieldValue($field_name, $value)
+    public function setCustomFieldValue($field_name, $value, $editor = "", $required = false)
     {
-        $this->custom_fields[$field_name] = $value;
+        if(empty($this->custom_fields)) $this->custom_fields = json_encode(array());
+        $array = json_decode($this->custom_fields, true);
+        if($required) {
```

Comment:
This logic block is incorrect. An editor parameter can be applied even if a required parameter is missing. So you can gut the IF block and just do:

`
$array[] = array(
    'name'     => $field_name,
    'value'    => $value,
    'editor'   => $editor,
    'required' => $required
);
`